api/app.py

```diff
@@ -81,36 +81,119 @@ def create_lister():
         return {"success": False, "message": "This username is taken!"}, STATUS_ALREADY_EXISTS
 
 
+def create_building(building_info):
+    conn = mysql.connection
+    cur = conn.cursor()
+
+    try:
+        cur.execute("INSERT INTO Building VALUES (NULL, %s, %s, %s, %s, %s)", 
+                    (building_info["address"], building_info["pet_friendly"], building_info["laundry_availability"], building_info["type_of_unit"], building_info["distance_from_waterloo"]))
+        conn.commit()
+
+        cur.execute("SELECT max(building_id) as building_id from building;")
```

Comment:
Use last_insert_id() instead of SELECT max(building_id). Can prevent wrong ids being returned if multiple users are writing. More info found [here](https://stackoverflow.com/questions/3180510/last-insert-id-vs-select-maxid) and [here](https://stackoverflow.com/questions/3837990/last-insert-id-mysql)